From c8ce547bea6461b3933a418c2c71b45b2049d939 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Tue, 9 Sep 2025 17:14:46 +0200
Subject: [PATCH] Import platform and adjust rtol for MP3 decoding

Added platform import and conditional rtol override for MP3 on Linux ARM64.
---
 test/test_encoders.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/test/test_encoders.py b/test/test_encoders.py
index f8b5b351..7d7a0271 100644
--- a/test/test_encoders.py
+++ b/test/test_encoders.py
@@ -1,6 +1,7 @@
 import io
 import json
 import os
+import platform
 import re
 import subprocess
 import sys
@@ -321,6 +322,10 @@ def test_against_cli(
             # TODO: https://github.com/pytorch/torchcodec/issues/837
             return
 
+        # Override rtol only for MP3 on Linux ARM64, see https://github.com/pytorch/torchcodec/issues/569#issuecomment-3197006256
+        if format == "mp3" and sys.platform == "linux" and platform.machine() == "aarch64":
+            rtol, atol = 0, 1e-2
+
         samples_by_us = self.decode(encoded_by_us)
         samples_by_ffmpeg = self.decode(encoded_by_ffmpeg)
 
